modules/internal/Bytes.chpl

```diff
@@ -64,10 +78,29 @@ module Bytes {
 
     :returns: A new :type:`bytes`
   */
+  @deprecated("createBytesWithBorrowedBuffer is deprecated - please use :proc:`bytes.createBorrowingBuffer` instead")
   inline proc createBytesWithBorrowedBuffer(x: c_string,
                                             length=x.size) : bytes {
-    return createBytesWithBorrowedBuffer(x:c_ptr(uint(8)), length=length,
-                                                           size=length+1);
+    return bytes.createBorrowingBuffer(x, length=length);
+  }
+
+  /*
+    Creates a new :type:`bytes` which borrows the internal buffer of a
+    `c_string`. If the buffer is freed before the :type:`bytes` returned
+    from this function, accessing it is undefined behavior.
+
+    :arg s: `c_string` to borrow the buffer from
+
+    :arg length: Length of `s`'s buffer, excluding the terminating
+                 null byte.
+    :type length: `int`
+
+    :returns: A new :type:`bytes`
+  */
+  inline proc type bytes.createBorrowingBuffer(x: c_string,
+                                            length=x.size) : bytes {
+    return bytes.createBorrowingBuffer(x:bufferType, length=length,
+                                          size=length+1);
```

Comment:
Can you align formals and actuals that fall to the second line to be in the same column as the first argument?

```
inline proc type bytes.createBorrowingBuffer(x: c_string,
                                             length=x.size) : bytes {
                                         //  ^
  return bytes.createBorrowingBuffer(x:bufferType, length=length,
                                     size=length+1);
                                 //  ^